internal/pkg/cmd/prerunner.go

```diff
@@ -181,6 +181,10 @@ func (r *PreRun) Authenticated(command *AuthenticatedCLICommand) func(*cobra.Com
 			return err
 		}
 
+		if r.Config.Context().GetCredentialType() == v1.APIKey {
+			return ErrIfMissingRunRequirement(cmd, r.Config)
```

Comment:
Returning here skips `r.setAuthenticatedContext`, which means `command.Context` is nil for api key logins; this causes issues if users supply the `--environment` or `--cluster` flags.